### PR TITLE
Feature(TEST): Test components functionality.

### PR DESCRIPTION
**Pull request by farouqdaif1**
### Feature(TEST): Test components functionality.
In this PR update:

-  Set up React Testing 
- Library.
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/unit/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    result = calculate(result, '5');
+    result = calculate(result, '=');
+    expect(result.total).toBe('15');
+  });
+  test('divide', () => {
+    let result = calculate(num, '10');
+    result = calculate(result, '÷');
+    result = calculate(result, '5');
+    result = calculate(result, '=');
+    expect(result.total).toBe('2');
```
Comment (mikenath223) [inline src/__tests__/unit/calculate.test.js:57]: - [x] Please write tests for other operators.

File: src/__tests__/unit/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    expect(result).toBe('2');
+  });
+  test('add', () => {
+    const result = operate('1', '2', '+');
+    expect(result).toBe('3');
+  });
+  test('modular', () => {
+    const result = operate('2', '2', '%');
+    expect(result).toBe('0');
+  });
```
Comment (mikenath223) [inline src/__tests__/unit/operate.test.js:15]: - [x] Write tests for other operators and also test division by zero.

File: src/__tests__/unit/calculate.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import calculate from '../../logic/calculate';
+
+describe('Validate calculate functionality', () => {
+  const num = { total: null, next: null, operation: null };
+  test('multiplication', () => {
+    let result = calculate(num, '10');
+    result = calculate(result, 'x');
+    result = calculate(result, '5');
+    result = calculate(result, '=');
+    expect(result.total).toBe('50');
```
Comment (mikenath223) [inline src/__tests__/unit/calculate.test.js:43]: - [x] Please use the React Testing library to stimulate user interaction on the calculator.

File: src/__tests__/unit/calculate.test.js (hunk 1 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:57]: Thanks for this awesome awake call. We initially thought a few would do. We will implement all the other pending tests.

> Cheers and HappyCoding 👨🏽‍💻

File: src/__tests__/unit/calculate.test.js (hunk 3 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:43]: Hello, @mikenath223 , What does this mean in context? We assumed did so. However, Encase we're missing a point, kindly do share more insight on what you mean.

Cheers 🥂

File: src/__tests__/unit/operate.test.js (hunk 2 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/operate.test.js:15]: >  On it 🏃🏽

File: src/__tests__/unit/calculate.test.js (hunk 1 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:57]: > Done ✅

File: src/__tests__/unit/operate.test.js (hunk 2 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/operate.test.js:15]: > Done ✅

File: src/__tests__/unit/calculate.test.js (hunk 3 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:43]: Actioned. ☑️

File: src/__tests__/unit/operate.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  test('divide', () => {
```
Comment (Mihndim2020) [inline src/__tests__/unit/operate.test.js:16]: - [x] As requested by the previous code reviewer, please kindly add a test for division by zero.

File: src/__tests__/unit/calculate.test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (Mihndim2020) [inline src/__tests__/unit/calculate.test.js:43]: - [x] As requested by the previous code reviewer, please kindly add tests to simulate user interactions using `React Testing Library`. For example, when you click on a particular set of button, what do you expect the page to render. Checkout this [article](https://www.codecademy.com/courses/learn-react-testing/lessons/react-testing-library/exercises/mimicking-user-interactions)

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there

 # Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/unit/operate.test.js (hunk 4 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/operate.test.js:16]: Hello @Mihndim2020 , This was not requested by the previous reviewer. Nevertheless, I'll do. Hopefully, the next reviewer does not suggest the changes you have not pointed out.

Thanks a lot && HappyCoding 👨🏽‍💻

Comment (blessedjasonmwanza) [reply src/__tests__/unit/operate.test.js:16]: Actioned.

File: src/__tests__/unit/calculate.test.js (hunk 5 above)
Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:43]: Hello once more @Mihndim2020 ,

You're recommending that I add this test in ```calculate.test.js``` test file right?
Kindly advise as the previous code reviewer took forever to respond to my question and it resulted in me making the same mistake.

Comment (blessedjasonmwanza) [reply src/__tests__/unit/calculate.test.js:43]: Hello @Mihndim2020 😃 ,

Thank you for sharing the linked article. I managed to implement the test.

Fingers crossed 🍀 🤞🏽  for the next review to carry an approval 😃 .

> HappyCoding

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @farouqdaif1 

Amazing job implementing suggestions from the previous reviewer! 💯 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
